### PR TITLE
Adding support of Jacoco in Maven dependencies

### PR DESCRIPTION
**Pull request by Isotech42**
I added the dependency of Jacoco in the `pom.xml`.
When you run `mvn test`, jacoco will do his job ^^
After that, if you launch the command `mvn jacoco:report`, a folder `target/site/jacoco` will be created containing a whole bunch of stuff indicating the code coverage of our tests.
There will be also an `index.html` file to see easily all the statistics.

Sources : 
- https://www.jacoco.org/jacoco/trunk/doc/maven.html
- https://www.jacoco.org/jacoco/trunk/doc/report-mojo.html

### CODE REVIEW

**Review by clemdiep-heig — APPROVED** [review]
Nice work!

We will have to add it to an automatic report with some other code inspection

File: pom.xml [hunk 1]
```diff
@@ -59,6 +59,64 @@
...
+          </execution>
+          <execution>
+            <id>report</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>report</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.target.directory}/jacoco</outputDirectory>
+            </configuration>
```
Comment (clemdiep-heig) [inline pom.xml:101]: I didn't see that but it create a directory with the name `${project.target.directory}` when executing `mvn clean install`. I tested and we can just remove those lines and it works fine.

```suggestion
```

**Review by clemdiep-heig — CHANGES_REQUESTED** [review]
Oupsi, I missed something

File: pom.xml (hunk 1 above)
Comment (Isotech42) [reply pom.xml:101]: Seriously ?? Ooooops, didn't see that

**Review by clemdiep-heig — APPROVED** [review]

**Review by dylanrodr-heig — APPROVED** [review]

**Pull request merged**